The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10204300776/job/28232516126) [(plain)](https://github.com/rust-lang/rust/commit/cbdc3778667c5bad3f9eee4124000a0ca96e4590/checks/28232516126/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#16 2.712 Building wheels for collected packages: reuse
#16 2.713   Building wheel for reuse (pyproject.toml): started
#16 2.959   Building wheel for reuse (pyproject.toml): finished with status 'done'
#16 2.960   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132715 sha256=dfa09868353292d98f811d3efdb0d54d07389e808efc71d68e3b93c514bf8bec
#16 2.960   Stored in directory: /tmp/pip-ephem-wheel-cache-5zze6xo4/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#16 2.962 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#16 3.359 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#16 3.359 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#16 DONE 3.4s
---
---- library/alloc/src/collections/btree/set.rs - collections::btree::set::BTreeSet<T,A>::lower_bound_mut (line 1243) stdout ----
error[E0308]: mismatched types
##[error]  --> library/alloc/src/collections/btree/set.rs:1252:32
   |
12 | assert_eq!(cursor.peek_prev(), Some(&mut 1));
   |                                ^^^^^^^^^^^^ types differ in mutability
   = note: expected enum `Option<&{integer}>`
              found enum `Option<&mut {integer}>`
              found enum `Option<&mut {integer}>`
help: try using `.as_deref()` to convert `Option<&mut {integer}>` to `Option<&{integer}>`
   |
12 | assert_eq!(cursor.peek_prev(), Some(&mut 1).as_deref());

error[E0308]: mismatched types
##[error]  --> library/alloc/src/collections/btree/set.rs:1253:32
   |
   |
13 | assert_eq!(cursor.peek_next(), Some(&mut 2));
   |                                ^^^^^^^^^^^^ types differ in mutability
   = note: expected enum `Option<&{integer}>`
              found enum `Option<&mut {integer}>`
              found enum `Option<&mut {integer}>`
help: try using `.as_deref()` to convert `Option<&mut {integer}>` to `Option<&{integer}>`
   |
13 | assert_eq!(cursor.peek_next(), Some(&mut 2).as_deref());

error[E0308]: mismatched types
##[error]  --> library/alloc/src/collections/btree/set.rs:1256:32
   |
   |
16 | assert_eq!(cursor.peek_prev(), Some(&mut 2));
   |                                ^^^^^^^^^^^^ types differ in mutability
   = note: expected enum `Option<&{integer}>`
              found enum `Option<&mut {integer}>`
              found enum `Option<&mut {integer}>`
help: try using `.as_deref()` to convert `Option<&mut {integer}>` to `Option<&{integer}>`
   |
16 | assert_eq!(cursor.peek_prev(), Some(&mut 2).as_deref());

error[E0308]: mismatched types
##[error]  --> library/alloc/src/collections/btree/set.rs:1257:32
   |
   |
17 | assert_eq!(cursor.peek_next(), Some(&mut 3));
   |                                ^^^^^^^^^^^^ types differ in mutability
   = note: expected enum `Option<&{integer}>`
              found enum `Option<&mut {integer}>`
              found enum `Option<&mut {integer}>`
help: try using `.as_deref()` to convert `Option<&mut {integer}>` to `Option<&{integer}>`
   |
17 | assert_eq!(cursor.peek_next(), Some(&mut 3).as_deref());

error[E0308]: mismatched types
##[error]  --> library/alloc/src/collections/btree/set.rs:1261:32
   |
   |
21 | assert_eq!(cursor.peek_next(), Some(&mut 1));
   |                                ^^^^^^^^^^^^ types differ in mutability
   = note: expected enum `Option<&{integer}>`
              found enum `Option<&mut {integer}>`
              found enum `Option<&mut {integer}>`
help: try using `.as_deref()` to convert `Option<&mut {integer}>` to `Option<&{integer}>`
   |
21 | assert_eq!(cursor.peek_next(), Some(&mut 1).as_deref());

error: aborting due to 5 previous errors

For more information about this error, try `rustc --explain E0308`.
For more information about this error, try `rustc --explain E0308`.
Couldn't compile the test.
---- library/alloc/src/collections/btree/set.rs - collections::btree::set::BTreeSet<T,A>::upper_bound_mut (line 1329) stdout ----
error[E0308]: mismatched types
##[error]  --> library/alloc/src/collections/btree/set.rs:1338:32
   |
12 | assert_eq!(cursor.peek_prev(), Some(&mut 3));
   |                                ^^^^^^^^^^^^ types differ in mutability
   = note: expected enum `Option<&{integer}>`
              found enum `Option<&mut {integer}>`
              found enum `Option<&mut {integer}>`
help: try using `.as_deref()` to convert `Option<&mut {integer}>` to `Option<&{integer}>`
   |
12 | assert_eq!(cursor.peek_prev(), Some(&mut 3).as_deref());

error[E0308]: mismatched types
##[error]  --> library/alloc/src/collections/btree/set.rs:1339:32
   |
   |
13 | assert_eq!(cursor.peek_next(), Some(&mut 4));
   |                                ^^^^^^^^^^^^ types differ in mutability
   = note: expected enum `Option<&{integer}>`
              found enum `Option<&mut {integer}>`
              found enum `Option<&mut {integer}>`
help: try using `.as_deref()` to convert `Option<&mut {integer}>` to `Option<&{integer}>`
   |
13 | assert_eq!(cursor.peek_next(), Some(&mut 4).as_deref());

error[E0308]: mismatched types
##[error]  --> library/alloc/src/collections/btree/set.rs:1342:32
   |
   |
16 | assert_eq!(cursor.peek_prev(), Some(&mut 2));
   |                                ^^^^^^^^^^^^ types differ in mutability
   = note: expected enum `Option<&{integer}>`
              found enum `Option<&mut {integer}>`
              found enum `Option<&mut {integer}>`
help: try using `.as_deref()` to convert `Option<&mut {integer}>` to `Option<&{integer}>`
   |
16 | assert_eq!(cursor.peek_prev(), Some(&mut 2).as_deref());

error[E0308]: mismatched types
##[error]  --> library/alloc/src/collections/btree/set.rs:1343:32
   |
   |
17 | assert_eq!(cursor.peek_next(), Some(&mut 3));
   |                                ^^^^^^^^^^^^ types differ in mutability
   = note: expected enum `Option<&{integer}>`
              found enum `Option<&mut {integer}>`
              found enum `Option<&mut {integer}>`
help: try using `.as_deref()` to convert `Option<&mut {integer}>` to `Option<&{integer}>`
   |
17 | assert_eq!(cursor.peek_next(), Some(&mut 3).as_deref());

error[E0308]: mismatched types
##[error]  --> library/alloc/src/collections/btree/set.rs:1346:32
   |
   |
20 | assert_eq!(cursor.peek_prev(), Some(&mut 4));
   |                                ^^^^^^^^^^^^ types differ in mutability
   = note: expected enum `Option<&{integer}>`
              found enum `Option<&mut {integer}>`
              found enum `Option<&mut {integer}>`
help: try using `.as_deref()` to convert `Option<&mut {integer}>` to `Option<&{integer}>`
   |
20 | assert_eq!(cursor.peek_prev(), Some(&mut 4).as_deref());

error: aborting due to 5 previous errors

For more information about this error, try `rustc --explain E0308`.
---
    library/alloc/src/collections/btree/set.rs - collections::btree::set::BTreeSet<T,A>::upper_bound_mut (line 1329)

test result: FAILED. 746 passed; 2 failed; 4 ignored; 0 measured; 0 filtered out; finished in 33.60s

error: doctest failed, to rerun pass `-p alloc --doc`
  local time: Thu Aug  1 19:28:38 UTC 2024
  network time: Thu, 01 Aug 2024 19:28:38 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/alloc/src/collections/btree/set.rs
+++ b/library/alloc/src/collections/btree/set.rs
@@ -1250,16 +1250,16 @@ impl<T, A: Allocator + Clone> BTreeSet<T, A> {
     /// let mut set = BTreeSet::from([1, 2, 3, 4]);
     ///
     /// let mut cursor = set.lower_bound_mut(Bound::Included(&2));
-    /// assert_eq!(cursor.peek_prev(), Some(&mut 1));
-    /// assert_eq!(cursor.peek_next(), Some(&mut 2));
+    /// assert_eq!(cursor.peek_prev(), Some(&1));
+    /// assert_eq!(cursor.peek_next(), Some(&2));
     ///
     /// let mut cursor = set.lower_bound_mut(Bound::Excluded(&2));
-    /// assert_eq!(cursor.peek_prev(), Some(&mut 2));
-    /// assert_eq!(cursor.peek_next(), Some(&mut 3));
+    /// assert_eq!(cursor.peek_prev(), Some(&2));
+    /// assert_eq!(cursor.peek_next(), Some(&3));
     ///
     /// let mut cursor = set.lower_bound_mut(Bound::Unbounded);
     /// assert_eq!(cursor.peek_prev(), None);
-    /// assert_eq!(cursor.peek_next(), Some(&mut 1));
+    /// assert_eq!(cursor.peek_next(), Some(&1));
     /// ```
     #[unstable(feature = "btree_cursors", issue = "107540")]
     pub fn lower_bound_mut<Q: ?Sized>(&mut self, bound: Bound<&Q>) -> CursorMut<'_, T, A>
@@ -1336,15 +1336,15 @@ impl<T, A: Allocator + Clone> BTreeSet<T, A> {
     /// let mut set = BTreeSet::from([1, 2, 3, 4]);
     ///
     /// let mut cursor = unsafe { set.upper_bound_mut(Bound::Included(&3)) };
-    /// assert_eq!(cursor.peek_prev(), Some(&mut 3));
-    /// assert_eq!(cursor.peek_next(), Some(&mut 4));
+    /// assert_eq!(cursor.peek_prev(), Some(&3));
+    /// assert_eq!(cursor.peek_next(), Some(&4));
     ///
     /// let mut cursor = unsafe { set.upper_bound_mut(Bound::Excluded(&3)) };
-    /// assert_eq!(cursor.peek_prev(), Some(&mut 2));
-    /// assert_eq!(cursor.peek_next(), Some(&mut 3));
+    /// assert_eq!(cursor.peek_prev(), Some(&2));
+    /// assert_eq!(cursor.peek_next(), Some(&3));
     ///
     /// let mut cursor = unsafe { set.upper_bound_mut(Bound::Unbounded) };
-    /// assert_eq!(cursor.peek_prev(), Some(&mut 4));
+    /// assert_eq!(cursor.peek_prev(), Some(&4));
     /// assert_eq!(cursor.peek_next(), None);
     /// ```
     #[unstable(feature = "btree_cursors", issue = "107540")]

